bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java

```diff
@@ -0,0 +1,550 @@
+                                                    ThingStatusDetail.CONFIGURATION_ERROR,
+                                                    "Vin is not available through this account.");
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    } else {
+                        if (response != null) {
+                            logger.error("Error fetching the list of vehicles : {}:{}", response.getStatus(),
+                                    response.getStatusInfo());
+                            updateStatus(ThingStatus.OFFLINE);
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("An exception occurred while connecting to the Tesla back-end: '{}'", e.getMessage());
+        } finally {
+            lock.unlock();
+        }
+    };
+
+    public static class Authenticator implements ClientRequestFilter {
+        private final String user;
+        private final String password;
+
+        public Authenticator(String user, String password) {
+            this.user = user;
+            this.password = password;
+        }
+
+        @Override
+        public void filter(ClientRequestContext requestContext) throws IOException {
+            MultivaluedMap<String, Object> headers = requestContext.getHeaders();
+            final String basicAuthentication = getBasicAuthentication();
+            headers.add("Authorization", basicAuthentication);
+        }
+
+        private String getBasicAuthentication() {
+            String token = this.user + ":" + this.password;
+            return "Basic " + Base64.getEncoder().encodeToString(token.getBytes(StandardCharsets.UTF_8));
+        }
+    }
+
+    protected class Request implements Runnable {
+
+        private TeslaVehicleHandler handler;
+        private String request;
+        private String payLoad;
+        private WebTarget target;
+
+        public Request(TeslaVehicleHandler handler, String request, String payLoad, WebTarget target) {
+            this.handler = handler;
+            this.request = request;
+            this.payLoad = payLoad;
+            this.target = target;
+        }
+
+        @Override
+        public void run() {
+            try {
+                String result = "";
+
+                if (getThing().getStatus() == ThingStatus.ONLINE) {
+                    result = invokeAndParse(handler.getVehicleId(), request, payLoad, target);
+                    if (result != null && !"".equals(result)) {
+                        handler.parseAndUpdate(request, payLoad, result);
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("An exception occurred while executing a request to the vehicle: '{}'", e.getMessage());
```

Comment:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.